File: src/main/java/io/github/roguesdev/hoard/rxjava2/DefaultRxDepositor.java
package io.github.roguesdev.hoard.rxjava2;

import io.github.roguesdev.hoard.ReactiveStreamDepositor;
import io.reactivex.Completable;
import io.reactivex.Single;

class DefaultRxDepositor<T> implements RxDepositor<T> {

  private final ReactiveStreamDepositor<T> depositor;

  DefaultRxDepositor(ReactiveStreamDepositor<T> depositor) {
    this.depositor = depositor;
  }

  @Override public Completable store(T value) {
    return Completable.fromPublisher(depositor.store(value));
  }

  @Override public Single<T> retrieve() {
    return Single.fromPublisher(depositor.retrieve());
  }

  @Override public Completable delete() {
    return Completable.fromPublisher(depositor.delete());
  }

  @Override public Single<Boolean> exists() {
    return Single.fromPublisher(depositor.exists());
  }
}

File: src/main/kotlin/io/github/roguesdev/hoard/rxjava2/DefaultRxDepositor.kt
package io.github.roguesdev.hoard.rxjava2

import io.github.roguesdev.hoard.ReactiveStreamDepositor
import io.reactivex.Completable
import io.reactivex.Single

internal class DefaultRxDepositor<T>(private val depositor: ReactiveStreamDepositor<T>) : RxDepositor<T> {

    override fun store(value: T): Completable {
        return Completable.fromPublisher(depositor.store(value))
    }

    override fun retrieve(): Single<T> {
        return Single.fromPublisher(depositor.retrieve())
    }

    override fun delete(): Completable {
        return Completable.fromPublisher(depositor.delete())
    }

    override fun exists(): Single<Boolean> {
        return Single.fromPublisher(depositor.exists())
    }
}
